jupyterlab_git/git.py

```diff
@@ -555,19 +555,30 @@ async def detailed_log(self, selected_hash, path):
         Execute git log -1 --numstat --oneline -z command (used to get
         insertions & deletions per file) & return the result.
         """
-        cmd = ["git", "log", "-1", "--numstat", "--oneline", "-z", selected_hash]
+        cmd = [
+            "git",
+            "log",
+            "-1",
+            "--numstat",
+            "--pretty=format:%b%x00",
+            "-z",
+            selected_hash,
+        ]
+
         code, my_output, my_error = await execute(
             cmd,
             cwd=path,
         )
-
         if code != 0:
             return {"code": code, "command": " ".join(cmd), "message": my_error}
 
         total_insertions = 0
         total_deletions = 0
         result = []
         line_iterable = iter(strip_and_split(my_output)[1:])
+
+        commit_body = my_output.split("\x00", 1)[0].strip()
+
         for line in line_iterable:
```

Comment:
As the output is split by `strip_and_split`, we can simplify further the code.

```suggestion
        line_iterable = strip_and_split(my_output)

        commit_body = line_iterable[0]

        for line in line_iterable[1:]:
```

Reply:
Hey @fcollonval,

I tried using this suggestion, but when I test it in the JupyterLab extension--the output looks like the following:

![image](https://user-images.githubusercontent.com/68607795/174364715-6c4266b3-b09c-497f-a418-786a4de7af14.png)

It does not seem to work for commits descriptions. It works fine for commits with a summary and description.

Reply:
Thanks Linda for trying. We are hit by the initial strip of `\x00` in front the string when there is no description.

Could you try the following a first split as you implemented; but using the second part to list the files:

```py
      first_split = my_output.split("\x00", 1)
      commit_body = first_split[0].strip()
      line_iterable = strip_and_split(first_split[1]))
      for line in line_iterable:
```

Reply:
Hi @fcollonval this worked, and I've pushed this change with this update.